### PR TITLE
Update tags for CoreHistogramView plugins

### PR DESCRIPTION
**Pull request by sol695510**
### Description of the Change
Updated tags for CoreHistogramView plugins.

### Alternate Designs
N/A

### Why Should This Be In Core?
N/A

### Benefits
N/A

### Possible Drawbacks
N/A

### Verification Process
N/A

### Applicable Issues
#1126

### CODE REVIEW

**Comment by arcturus2** [comment]
I'm not sure on this one and keen to have a discussion. The histogram filtering is pretty powerful so I would somehow like it to not be "LOW LEVEL". I say this because if I was to use the tags for metrics I would probably ignore the "LOW LEVEL" ones as they are more "not user driven". The histogram filtering is definitely something a user has to manually do so I was thinking either "SELECTION" or "MODIFY" ?

**Comment by sol695510** [comment]
> I'm not sure on this one and keen to have a discussion. The histogram filtering is pretty powerful so I would somehow like it to not be "LOW LEVEL". I say this because if I was to use the tags for metrics I would probably ignore the "LOW LEVEL" ones as they are more "not user driven". The histogram filtering is definitely something a user has to manually do so I was thinking either "SELECTION" or "MODIFY" ?

I was unsure about this one too but since it was filtering I left it as LOW LEVEL. Thanks for the clarification, I'll make the changes.

**Comment by aldebaran30701** [comment]
Out of SELECTION and MODIFY, I'd have to go with MODIFY as feeling right. It does edit a graph attribute after all.

**Comment by antares1470** [comment]
> I'm not sure on this one and keen to have a discussion. The histogram filtering is pretty powerful so I would somehow like it to not be "LOW LEVEL". I say this because if I was to use the tags for metrics I would probably ignore the "LOW LEVEL" ones as they are more "not user driven". The histogram filtering is definitely something a user has to manually do so I was thinking either "SELECTION" or "MODIFY" ?

@arcturus2 @sol695510 I know the changes have already been made but I think LOW LEVEL is actually the appropriate tag in this instance. Looking at the code for the two histogram filter plugins in question (`HistogramClearFilterPlugin` and `HistogramFilterOnSelectionPlugin`), all they are doing is updating the state meta attribute. Based on how state updaters have been tagged elsewhere, LOW LEVEL should be what's added here

**Comment by arcturus2** [comment]
That is true @antares1470 and I agree with you on that basis. @sol695510 given I got you to make the change and now asking if you can roll it back, I am happy to made the change to your PR and get this one merged in. Let me know.

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java [hunk 1]
```diff
@@ -588,7 +588,7 @@ public String getName() {
     /**
      * Plugin to only select bins.
      */
-    @PluginInfo(pluginType = PluginType.NONE, tags = {"MODIFY"})
+    @PluginInfo(pluginType = PluginType.UPDATE, tags = {"SELECT"})
```
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java:591]: Maybe go with `PluginType.SELECTION` if changing tag to select

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java [hunk 2]
```diff
@@ -616,7 +616,7 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
     /**
      * Plugin to select bins.
      */
-    @PluginInfo(pluginType = PluginType.NONE, tags = {"MODIFY"})
+    @PluginInfo(pluginType = PluginType.UPDATE, tags = {"SELECT"})
```
Comment (antares1470) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java:619]: similar

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by antares1470 — APPROVED** [review]

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java [hunk 3]
```diff
@@ -637,7 +581,201 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
...
+            if (selectedAttribute != Graph.NOT_FOUND) {
+                currentSelectedModificationCount = graph.getValueModificationCounter(selectedAttribute);
+            }
+        }
+    }
+
+    /**
+     * Plugin to invert bins.
+     */
+    @PluginInfo(pluginType = PluginType.UPDATE, tags = {"MODIFY"})
```
Comment (arcturus2) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java:649]: @sol695510 curios to know the reason you went with `MODIFY`. I was thinking `SELECTION` as according to the code it looks like the bins selected would change so the end result is just a selection change. I was however unable to find the "option" in the Histogram to trigger this.

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java [hunk 4]
```diff
@@ -637,7 +581,201 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
...
+            if (selectedAttribute != Graph.NOT_FOUND) {
+                currentSelectedModificationCount = graph.getValueModificationCounter(selectedAttribute);
+            }
+        }
+    }
+
+    /**
+     * Plugin to complete bins.
+     */
+    @PluginInfo(pluginType = PluginType.UPDATE, tags = {"MODIFY"})
```
Comment (arcturus2) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java:677]: Ditto `SELECTION`? Again keen to get your feedback on it @sol695510.

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java [hunk 5]
```diff
@@ -637,7 +581,201 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
...
+            if (selectedAttribute != Graph.NOT_FOUND) {
+                currentSelectedModificationCount = graph.getValueModificationCounter(selectedAttribute);
+            }
+        }
+    }
+
+    /**
+     * Plugin to filter selection.
+     */
+    @PluginInfo(pluginType = PluginType.UPDATE, tags = {"MODIFY"})
```
Comment (arcturus2) [inline CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java:705]: Ditto `SELECTION`

**Review by arcturus2 — COMMENTED** [review]
Put some feedback for you @sol695510

File: CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java (hunk 3 above)
Comment (sol695510) [reply CoreHistogramView/src/au/gov/asd/tac/constellation/views/histogram/HistogramTopComponent.java:649]: I'm looking at it again now and can't figure why I did PluginType.UPDATE and MODIFY, I've changed them to PluginType.SELECTION and SELECT. Same for the other two.

**Review by arcturus2 — APPROVED** [review]

**Pull request merged**